### PR TITLE
Add multi targeting to projects

### PR DESCRIPTION
**Pull request by cwensley**
- Support compiling with WinForms in .NET 7 on Mac
- Add VS debug targets for Rhino 8 netfx/netcore
- Include launch.json/tasks.json for VS Code to make it easier to start debugging on Mac
- Remove VS for Mac build
- Adds a target to build .yak packages for Rhino 7 and Rhino 8 if specified.

Fixes #16

### CODE REVIEW

File: Rhino.Templates/content/CSGrasshopper/.vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+      "preLaunchTask": "build",
+      "program": "",
+      "osx": {
+        "program": "/Applications/Rhino 8.app/Contents/MacOS/Rhinoceros"
+      },
+      "windows": {
+        "program": "C:\\Program Files\\Rhino 8\\System\\Rhino.exe",
+        "targetArchitecture": "x86_64",
+      },
+      "args": [],
```
Comment (clicketyclackety) [inline Rhino.Templates/content/CSGrasshopper/.vscode/launch.json:17]: I would love it if Grasshopper opens when I debug a GH Plugin
```suggestion
      "osx": {
            "args": ["-runscript=Grasshopper"],
      },
      "windows": {
            "args": ["/runscript=\"Grasshopper\""],
      },
```

I'm quite sure the args is correct, but I've never used an osx/windows condition? Would the above work?

**Review by clicketyclackety — CHANGES_REQUESTED** [review]
# Other Questions
1. Any reason not to include a `settings.json`
``` json
{
        "dotnet.defaultSolution": "MySolutionName.sln"
}
```
2. Should we include a build netfx task?
3. When using `dotnet new rhinocommand --help` I see quite a few options. `-lang` and `-type` would be good to include possible options as I wasn't sure what options were available until I entered a wrong value by running `dotnet new rhinocommand -lang --help` the same way `--version` is done `--version <6|7|8>` would be great
4. I like `-sample`, awesome to find that! I'll include that in the guide
5. Everything else I've tried works.
6. Everything builds on both platforms (Besides ZooManager on mac)

File: Rhino.Templates/content/CSGrasshopper/.vscode/launch.json (hunk 1 above)
Comment (cwensley) [reply Rhino.Templates/content/CSGrasshopper/.vscode/launch.json:17]: These args should be added to the existing "osx" and "windows" sections, not create new ones just for args.  Yes it should work, though I think on Windows it should be "/runscript=_Grasshopper", and macOS "-runscript=_Grasshopper", with underscores so it is language agnostic.  This should also be applied to the VBGrasshopper template too.

File: Rhino.Templates/content/CSGrasshopper/MyGrasshopper.1.csproj [hunk 2]
```diff
@@ -1,25 +1,119 @@
...
+    <Warning Text="Could not find Yak executable" Condition="$(YakExecutable) == ''" />
+
+    <ItemGroup>
+      <YakPackagesToDelete Include="$(OutputPath)\*.yak;$(OutputPath)\**\manifest.yml" />
+    </ItemGroup>
+    
+    <Delete Files="@(YakPackagesToDelete)" />
+
+    <Exec Command="&quot;$(YakExecutable)&quot; spec" WorkingDirectory="$(OutputPath)" Condition="$(BuildYakPackage) == 'True'" />
+    <Exec Command="&quot;$(YakExecutable)&quot; build" WorkingDirectory="$(OutputPath)" Condition="$(BuildYakPackage) == 'True'" />
```
[the remark this answers is not in the document]
Comment (cwensley) [reply Rhino.Templates/content/CSGrasshopper/MyGrasshopper.1.csproj:104]: Yeah I had thoughts of supporting a custom manifest.yml.  Ideally this target can be added to a separate nuget package perhaps "RhinoYakPackage" so we can simplify the projects and also have a better "out of the box" experience.  I initially thought of adding to RhinoCommon nuget package but that wouldn't work with older rhino versions (e.g. if you want to target RhinoCommon 8.0 for example).

**Comment by cwensley** [comment]
> # Other Questions
> 1. Any reason not to include a `settings.json`
> 
> ```json
> {
>         "dotnet.defaultSolution": "MySolutionName.sln"
> }
> ```

We don't include a .sln file, and the C# VSCode extension automatically will pick up the .csproj and/or .sln and recent builds even generates a .sln for you.  

>  2. Should we include a build netfx task?

Why _build_ netfx?  It already builds all tfm's, net45, net48, net7.0 if specified.  Do you mean a launch.json config with netfx on Windows?  If so, yes we could to match what we have in the VS launchSettings.json.

> 3. When using `dotnet new rhinocommand --help` I see quite a few options. `-lang` and `-type` would be good to include possible options as I wasn't sure what options were available until I entered a wrong value by running `dotnet new rhinocommand -lang --help` the same way `--version` is done `--version <6|7|8>` would be great

-lang and -type are built-in to dotnet new (you will see it in the "Options" section when using --help).  I'm not sure we can add any additional help text to it.

> 4. I like `-sample`, awesome to find that! I'll include that in the guide

👍

> 5. Everything else I've tried works.

Yay!

> 6. Everything builds on both platforms (Besides ZooManager on Mac)

Yep, Zoo is Windows only currently.

**Review by clicketyclackety — APPROVED** [review]

**Review by clicketyclackety — APPROVED** [review]

**Pull request merged**